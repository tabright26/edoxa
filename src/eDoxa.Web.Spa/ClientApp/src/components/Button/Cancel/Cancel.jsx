import React from "react";
import { Button } from "reactstrap";

const CancelButton = ({ className, onClick, width = "75px" }) => (
  <Button className={className} style={{ width }} color="primary" outline size="sm" type="button" onClick={onClick}>
    Cancel
  </Button>
);

export default CancelButton;
