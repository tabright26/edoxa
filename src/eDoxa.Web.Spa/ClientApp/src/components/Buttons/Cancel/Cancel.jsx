import React from "react";
import { Button } from "reactstrap";

const CancelButton = ({ className, onClick }) => (
  <Button className={className} color="primary" outline size="sm" type="button" onClick={onClick}>
    Cancel
  </Button>
);

export default CancelButton;
