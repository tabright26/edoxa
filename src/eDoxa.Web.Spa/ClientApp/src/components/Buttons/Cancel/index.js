import CancelButton from "./Cancel";

export default CancelButton;
