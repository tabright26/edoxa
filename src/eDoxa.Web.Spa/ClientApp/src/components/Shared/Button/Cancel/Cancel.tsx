import React, { FunctionComponent } from "react";
import { Button } from "reactstrap";

const CancelButton: FunctionComponent<any> = ({ className, onClick, width = "75px" }) => (
  <Button className={className} style={{ width }} color="primary" outline size="sm" type="button" onClick={onClick}>
    Cancel
  </Button>
);

export default CancelButton;
