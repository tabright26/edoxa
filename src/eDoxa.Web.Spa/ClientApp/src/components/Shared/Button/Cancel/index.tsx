import React, { FunctionComponent } from "react";
import { Button } from "reactstrap";

interface Props {
  className?: string;
  size?: string;
  onClick?: () => void;
}

export const Cancel: FunctionComponent<Props> = ({
  className = null,
  onClick = null,
  size = null
}) => (
  <Button
    className={className}
    color="primary"
    size={size}
    outline
    onClick={onClick}
  >
    Cancel
  </Button>
);
