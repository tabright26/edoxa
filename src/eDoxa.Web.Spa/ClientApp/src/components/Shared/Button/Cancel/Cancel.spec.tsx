import React from "react";
import renderer from "react-test-renderer";
import Cancel from "./Cancel";

it("renders without crashing", () => {
  const tree = renderer.create(<Cancel />).toJSON();
  expect(tree).toMatchSnapshot();
});
