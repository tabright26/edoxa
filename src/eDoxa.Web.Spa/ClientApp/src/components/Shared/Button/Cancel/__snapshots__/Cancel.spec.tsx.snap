// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<button
  aria-label={null}
  className="btn btn-outline-primary btn-sm"
  onClick={[Function]}
  style={
    Object {
      "width": "75px",
    }
  }
  type="button"
>
  Cancel
</button>
`;
